### PR TITLE
feat(db) implement the db:cluster_mutex() for postgres strategy

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Implements db:cluster_mutex() for postgres strategy that was already implemented for Cassandra strategy.

### CODE REVIEW

File: kong/db/strategies/postgres/connector.lua [hunk 1]
```diff
@@ -357,6 +365,124 @@ function _mt:truncate_table(table_name)
...
+                       "       VALUES (", key_escaped, ", ", owner_escaped, ", ", ttl_escaped, ");\n",
+                       "COMMIT;"
+  }
+
+  local res, err = self:query(sql)
+  if not res then
+    return nil, err
+  end
+
+  if err ~= 4 then
```
[the remark this answers is not in the document]
Comment (bungle) [reply kong/db/strategies/postgres/connector.lua]: Naming `err` as `num_queries` is confusing on above lines. This is More of pgmoon design choices.

File: kong/db/strategies/postgres/connector.lua [hunk 2]
```diff
@@ -357,6 +365,124 @@ function _mt:truncate_table(table_name)
...
+  local owner_escaped
+  if owner == nil or owner == null then
+    owner_escaped = "NULL"
+  else
+    owner_escaped = self:escape_literal(owner)
+  end
+
+  local ttl_escaped
+  if ttl == nil or ttl == null then
+    ttl_escaped = concat { "TO_TIMESTAMP(", self:escape_literal(now_updated() + 60), ") AT TIME ZONE 'UTC'" }
```
[the remark this answers is not in the document]
Comment (bungle) [reply kong/db/strategies/postgres/connector.lua]: Cassandra had on setup call default valuen for ttl. I think that needs to Be removed then.

File: kong/db/strategies/postgres/connector.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (thibaultcha) [reply kong/db/strategies/postgres/connector.lua]: I meant the `4`, not `err` - why 4? Using a named constant would be more insightful. `err` is fine (I recall that about pgmoon), but could use a comment. I see you renamed it anyway.

File: kong/db/strategies/postgres/connector.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (thibaultcha) [reply kong/db/strategies/postgres/connector.lua]: Yes, Cassandra specifies a default at setup time and a value can also be specified at insert time. But what I meant here is that `db:cluster_mutex()` always sets a TTL before calling the underlying strategy implementation.

**Comment by thibaultcha** [comment]
Merged!

**Pull request closed without merging**